### PR TITLE
Update endpoints for NEGs from non-default subnet.

### PR DESCRIPTION
**Pull request by sawsa307**
/assign @gauravkghildiyal

### CODE REVIEW

File: pkg/neg/syncers/transaction_test.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func TestTransactionSyncNetworkEndpoints(t *testing.T) {
 				t.Errorf("Unexpected neg %q, expected %q", neg.Name, transactionSyncer.NegName)
 			}
 			if neg.NetworkEndpointType != string(testNegType) {
-				t.Errorf("Unexpected neg type %q, expected %q", neg.Type, testNegType)
+				t.Errorf("Unexpected neg type %q, expected %q", neg.NetworkEndpointType, testNegType)
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/transaction_test.go:130]: Thanks for the fix!

File: pkg/neg/syncers/transaction_test.go [hunk 2]
```diff
@@ -258,6 +258,237 @@ func TestTransactionSyncNetworkEndpoints(t *testing.T) {
...
+
+		testCases := []struct {
+			desc            string
+			addEndpoints    map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet
+			removeEndpoints map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet
+			expectEndpoints map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet
+		}{
+			{
+				"empty input",
+				map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet{},
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/transaction_test.go:354]: Can we please use the names of the fields -- hugely improves the readability.

File: pkg/neg/syncers/transaction_test.go [hunk 3]
```diff
@@ -258,6 +258,237 @@ func TestTransactionSyncNetworkEndpoints(t *testing.T) {
...
+					{Zone: testZone2, Subnet: additionalTestSubnet}: negtypes.NewNetworkEndpointSet().Union(generateEndpointSet(net.ParseIP("1.2.2.1"), 10, testInstance6, targetPort)),
+				},
+				map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet{},
+				map[negtypes.EndpointGroupInfo]negtypes.NetworkEndpointSet{
+					{Zone: testZone1, Subnet: defaultTestSubnet}:    negtypes.NewNetworkEndpointSet().Union(generateEndpointSet(net.ParseIP("1.1.2.1"), 10, testInstance2, targetPort)),
+					{Zone: testZone2, Subnet: defaultTestSubnet}:    negtypes.NewNetworkEndpointSet().Union(generateEndpointSet(net.ParseIP("1.1.4.1"), 10, testInstance4, targetPort)),
+					{Zone: testZone1, Subnet: additionalTestSubnet}: negtypes.NewNetworkEndpointSet().Union(generateEndpointSet(net.ParseIP("1.2.1.1"), 10, testInstance5, targetPort)),
+					{Zone: testZone2, Subnet: additionalTestSubnet}: negtypes.NewNetworkEndpointSet().Union(generateEndpointSet(net.ParseIP("1.2.2.1"), 10, testInstance6, targetPort)),
+				},
+			},
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/transaction_test.go:441]: Can we also add a test to exercise the removal of NEGs in non-default subnet

File: pkg/neg/syncers/transaction_test.go (hunk 2 above)
Comment (sawsa307) [reply pkg/neg/syncers/transaction_test.go:354]: Updated, thank you!

File: pkg/neg/syncers/transaction_test.go (hunk 3 above)
Comment (sawsa307) [reply pkg/neg/syncers/transaction_test.go:441]: Added, thanks!

**Review by sawsa307 — COMMENTED** [review]
@gauravkghildiyal Addressed all comments, ready for another round of review, thanks!

**Review by gauravkghildiyal — COMMENTED** [review]
/lgtm

Thanks!

**Comment by gauravkghildiyal** [comment]
Needs rebase

**Comment by sawsa307** [comment]
> Needs rebase

Rebased, thanks!

**Pull request merged**